controllers/integration/integration_adapter.go

```diff
@@ -189,15 +193,24 @@ func (a *Adapter) getAllApplicationComponents(application *hasv1alpha1.Applicati
 }
 
 // getAllApplicationReleaseLinks returns the ReleaseLinks used by the application being processed. If matching
-// ReleaseLinks are not found, an error will be returned.
+// ReleaseLinks are not found, an error will be returned. A ReleaseLink will only be returned if it has the
+// release.appstudio.openshift.io/auto-release label set to true or if it is missing the label entirely.
 func (a *Adapter) getAllApplicationReleaseLinks(application *hasv1alpha1.Application) (*[]releasev1alpha1.ReleaseLink, error) {
 	releaseLinks := &releasev1alpha1.ReleaseLinkList{}
-	opts := []client.ListOption{
-		client.InNamespace(application.Namespace),
-		client.MatchingFields{"spec.application": application.Name},
+	labelSelector := labels.NewSelector()
+	rlReq, err := labels.NewRequirement("release.appstudio.openshift.io/auto-release", selection.DoesNotExist, []string{"false"})
```

Comment:
This won’t work. When using DoesNotExist, the value passed must be empty. Maybe you can add multiple requirements to the selector, but I’m not sure about the logical expression used in those cases. I’d say that you can use NotIn with that value, to check that it’s not set to false. I’m assuming that if the label is not present, this resolve to true as well.